src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -649,22 +649,33 @@ public void WriteReferences(Dictionary<string, Dictionary<Slice, HashSet<Slice>>
 
                 foreach (var collections in referencesByCollection)
                 {
-                    var collectionTree = tx.InnerTransaction.CreateTree(_referenceCollectionPrefix + collections.Key); // #collection
```

Comment:
This is called when handling references in the `Map`, right?

Reply:
Yes. After all workers have done their work.